lib/bin/lwm2m_carrier/doc/app_integration.rst

```diff
@@ -69,105 +87,114 @@ See :file:`nrf\\lib\\bin\\lwm2m_carrier\\include\\lwm2m_carrier.h` for all the e
+
+* :c:macro:`LWM2M_CARRIER_EVENT_BOOTSTRAPPED`:
+
+  * This event indicates that the bootstrap sequence is complete, and that the device is ready to be registered.
+  * This event is typically seen during the first boot-up.
+
+* :c:macro:`LWM2M_CARRIER_EVENT_LTE_READY`:
+
+  * This event indicates that the device is registered to the LTE network (home or roaming).
+  * The bootstrap sequence is complete, and the application can use the LTE link.
+
+* :c:macro:`LWM2M_CARRIER_EVENT_REGISTERED`:
+
+  * This event indicates that the device has registered successfully to the carrier's device management servers.
+
+* :c:macro:`LWM2M_CARRIER_EVENT_DEFERRED`:
+
+  * This event indicates that the connection to the device management server has failed.
+  * The :c:macro:`LWM2M_CARRIER_EVENT_DEFERRED` event appears instead of the :c:macro:`LWM2M_CARRIER_EVENT_REGISTERED` event.
+  * The :c:member:`timeout` parameter supplied with this event determines when the LwM2M carrier library will retry the connection.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_NO_REASON`:
+
+  * The application need not take any special action.
+  * If :c:member:`timeout` is 24 hours, the application can proceed with other activities until the retry takes place.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_PDN_ACTIVATE`:
+
+  * This event indicates problem with the SIM card, or temporary network problems.
+  * If this persists, contact your carrier.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_BOOTSTRAP_CONNECT`:
+
+  * The DTLS handshake with the bootstrap server has failed.
+  * If the application is using a custom PSK, verify that the format is correct.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_BOOTSTRAP_SEQUENCE`:
+
+  * The bootstrap sequence is incomplete.
+  * The server failed either to acknowledge the request by the library, or to send objects to the library.
+  * Confirm that the carrier is aware of the IMEI.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_SERVER_NO_ROUTE`, :c:macro:`LWM2M_CARRIER_DEFERRED_BOOTSTRAP_NO_ROUTE`:
+
+  * There is a routing problem in the carrier network.
+  * If this event persists, contact the carrier.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_SERVER_CONNECT`:
+
+  * This event indicates that the DTLS handshake with the server has failed.
+  * This typically happens if the bootstrap sequence has failed on the carrier side.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_SERVER_REGISTRATION`:
+
+  * The server registration has not completed, and the server does not recognize the connecting device.
+  * If this event persists, contact the carrier.
+
+* :c:macro:`LWM2M_CARRIER_EVENT_FOTA_START`:
+
+  * This event indicates that the modem update has started.
+  * The application must immediately terminate any open TLS sessions.
+  * See :ref:`req_appln_limitations`.
+
+* :c:macro:`LWM2M_CARRIER_EVENT_REBOOT`:
+
+  * If the application is not ready to reboot, it must return non-zero and then reboot at the earliest convenient time.
```

Comment:
I know the event is quite self-explanatory, but maybe it would make sense to add an event description here as well (to be consistent with the other events and follow the description -> notes pattern)?